router.go

```diff
@@ -158,13 +158,15 @@ func New() *Router {
 // Group returns a new grouped Router.
 // Path auto-correction, including trailing slashes, is enabled by default.
 func (r *Router) Group(path string) *Router {
-	return &Router{
+	g := &Router{
 		beginPath:              path,
 		RedirectTrailingSlash:  true,
 		RedirectFixedPath:      true,
 		HandleMethodNotAllowed: true,
 		HandleOPTIONS:          true,
 	}
+	r.NotFound = g.Handler
```

Comment:
This is valid to create just one group, but if you create a second or more groups, the previous will be unregistered, because you are overwrite the general NotFound handler, you must overwrite the NotFound handler of group router after register the first group.

```go
r := New()
r.GET("/", index)

v1 := r.Group("/v1")
v1.GET("/foo", fooHandler)
r.NotFound = v1.Handler

v2 := r.Group("/v2")
v2.GET("/bar", barHandler)
v1.NotFound = v2.Handler

...
```

Reply:
Thanks @savsgio, for catching the bug there. I have pushed a fix.

Reply:
But i think it's a slow solution because to access to group routes must be process the handler 
as many times as groups have registered until you find it.

So I propose to create groups like now, but when you add new routes inside the group, registered these paths to main router (group path + subpath: "/v1" + "/foo") and not chain groups with NotFoundHanlder.

Reply:
The updated code seems a bit more performant. Mostly same as regular routing. However, this route registration runs only when the registration starts. It only run once.

Reply:
I've been testing your code, and subgroups are unrecheable.
```go
r := New()

v1 := r.Group("/v1")
v1.GET("/foo", Index)

data := v1.Group("/data")  // Subgroup of "/v1"  --> "/v1/data/bar"
data.GET("/bar", Index)
```

I could resolve that modifying the `Router.Handle`
```go
func (r *Router) Handle(method, path string, handle fasthttp.RequestHandler) {
	if path[0] != '/' {
		panic("path must begin with '/' in path '" + path + "'")
	}
	if r.beginPath != "/" {
		path = r.beginPath + path
	}
 
        // Call to the parent recursively until main router to register paths in it
	if r.parent != nil {
		r.parent.Handle(method, path, handle)
		return
	}

	if r.trees == nil {
		r.trees = make(map[string]*node)
	}

	root := r.trees[method]
	if root == nil {
		root = new(node)
		r.trees[method] = root
	}

	root.addRoute(path, handle)
}
```

Reply:
if you could fix it, i will merge the PR

Reply:
Again, this fixed in gramework a long time ago, we can just get the solution there :)

Reply:
Fixed! and tested as well 😄

Reply:
Thanks!

Reply:
> Again, this fixed in gramework a long time ago, we can just get the solution there :)

I know, but for a while this is a simple solution without loss performance in hot paths! 😄

Reply:
> this is a simple solution without loss performance in hot paths! 

@savsgio I have exactly the same recursive mechanics :) this particular implementation just skips that self-documenting feature like *SubRouter, but otherwise it's identical :)

Reply:
Ahhh ok nice! hehe, later we could adds `*SubRouter`feature, but now the router chaining is supported. I think it's a similar feature, right?